File: src/dotnet/Common/Models/ResourceProviders/AIModel/AIModelBase.cs
using FoundationaLLM.Common.Constants.ResourceProviders;
using FoundationaLLM.Common.Models.ResourceProviders.DataSource;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.ResourceProviders.AIModel
{
    /// <summary>
    /// Base model type for AIModel resources
    /// </summary>
    [JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
    [JsonDerivedType(typeof(EmbeddingAIModel), AIModelTypes.Embedding)]
    [JsonDerivedType(typeof(AzureDataLakeDataSource), DataSourceTypes.AzureDataLake)]
    public class AIModelBase : ResourceBase
    {
        /// <inheritdoc/>
        [JsonIgnore]
        public override string? Type { get; set; }
        /// <summary>
        /// The endpoint metadata needed to call the AI model endpoint
        /// </summary>
        public AIModelEndpoint? Endpoint { get; set; }
        /// <summary>
        /// The version for the AI model
        /// </summary>
        public string? Version { get; set; }
        /// <summary>
        /// Deployment name for the AI model
        /// </summary>
        public string? DeploymentName { get; set; }
        /// <summary>
        /// Key value parameters configured for the model
        /// </summary>
        public Dictionary<string, object> ModelParameters { get; set; } = new Dictionary<string, object>();

    }
}

File: src/python/PythonSDK/foundationallm/models/resource_providers/ai_models/ai_model_base.py
from pydantic import Field
from typing import Any, Self, Optional
from foundationallm.models.resource_providers import ResourceBase
from foundationallm.utils import ObjectUtils
from foundationallm.langchain.exceptions import LangChainException

class AIModelBase(ResourceBase):
    """
    The base class used for AIModel resources.
    """
    endpoint_object_id: str = Field(description="The object ID of the APIEndpointConfiguration object providing the configuration for the API endpoint used to interact with the model.")
    version: Optional[str] = Field(description="The version of the AI model.")
    deployment_name: Optional[str] = Field(description="The deployment name for the AI model.")
    model_parameters: Optional[dict] = Field(default={}, description="A dictionary containing default values for model parameters.")

    @staticmethod
    def from_object(obj: Any) -> Self:

        ai_model: AIModelBase = None

        try:
            ai_model = AIModelBase(**ObjectUtils.translate_keys(obj))
        except Exception as e:
            raise LangChainException(f"The AI model object provided is invalid. {str(e)}", 400)
        
        if ai_model is None:
            raise LangChainException("The AI model object provided is invalid.", 400)

        return ai_model
